### PR TITLE
Add Operating System filter

### PR DESCRIPTION
**Pull request by kruai**
This addresses [ESSNTL-1405](https://issues.redhat.com/browse/ESSNTL-1405). It allows filtering on the `system_profile.operating_system` object. Note that the list of operating systems in the filter is currently a hardcoded list, like how Advisor does it; we can update this later once we implement an endpoint that gets all the unique values for a field ([ESSNTL-564](https://issues.redhat.com/browse/ESSNTL-564)).

### CODE REVIEW

**Comment by karelhala** [comment]
You can use the new treeview filter in the conditional filter. It would look like - http://patternfly-react.surge.sh/components/tree-view/#with-checkboxes. I think you can continue with this work and we can refactor once this is done.

**Comment by kruai** [comment]
> You can use the new treeview filter in the conditional filter. It would look like - http://patternfly-react.surge.sh/components/tree-view/#with-checkboxes. I think you can continue with this work and we can refactor once this is done.

Sure, I'd be happy to make that a follow-up after we get the basic implementation in!
Also @karelhala this is a shot in the dark, but I'm having trouble nailing down where the `[0]` is coming from in the generated search params (CC @ryandillinfelton). For instance, when I select "RHEL 8.3" in the filter, it generates the query param `&filter[system_profile][operating_system][RHEL][version][0]=8.3`. I can go back to debugging this after some other tasks, but I wanted to check with y'all to see if you remember dealing with something like this before.

**Comment by kruai** [comment]
@karelhala @ryandillinfelton Do either of you see a reason the `[0]` is getting appended to the query param? That's the main thing blocking me right now that I can't figure out :disappointed:

**Review by karelhala — APPROVED** [review]
Just a few minor nit picks but overall it looks really solid!

File: src/Utilities/constants.js [hunk 1]
```diff
@@ -17,6 +18,125 @@ export const defaultFilters = {
     registeredWithFilter: ['insights']
 };
 
+export const operatingSystems = [
```
[the remark this answers is not in the document]
Comment (kruai) [reply src/Utilities/constants.js:21]: Yep, it's the one I put in the description: https://issues.redhat.com/browse/ESSNTL-564

File: src/Utilities/constants.js [hunk 2]
```diff
@@ -64,11 +184,16 @@ export function reduceFilters(filters = []) {
                 ...acc,
                 staleFilter: oneFilter.staleFilter
             };
-        } else if ('registeredWithFilter' in oneFilter) {
+        }  else if ('registeredWithFilter' in oneFilter) {
             return {
                 ...acc,
                 registeredWithFilter: oneFilter.registeredWithFilter
             };
+        } else if ('osFilter' in oneFilter) {
```
[the remark this answers is not in the document]
Comment (kruai) [reply src/Utilities/constants.js]: OK, I've updated it!

**Review by karelhala — APPROVED** [review]

**Comment by kruai** [comment]
I just discovered that the `filter` object doesn't yet support multiple values for the same field because it's interpreted as a JSON object, and the keys overwrite each other. I'm working on a fix for that now, and will have to make additional changes to this PR.

**Comment by kruai** [comment]
Actually, I didn't realize that it already handled that scenario. In that case, this can be merged whenever the API side is merged and ready

**Comment by kruai** [comment]
The [API-side PR](https://github.com/RedHatInsights/insights-host-inventory/pull/1045) has been merged, so I'm merging this one as well. Heads up that it should not be pushed to Prod until the next API prod deploy (which will hopefully be very soon).

**Pull request merged**